### DISCUSSION TITLE
Thymeleaf Form binding

### DISCUSSION THREAD

**Discussion by pellerito** [thymeleaf/thymeleaf#1007] [Q&A]
In Thymeleaf I have this situation:

`<select class="form-select input-widget" id="treat" tabindex="31" th:field="*{treat}">
    <option value="">Please select...</option>
    <option th:each="v : ${@baba.getTreat()}" th:value="${v.key}" th:text="${v.value}"></option>
</select>`

and I have

`String treat;`

and i.e.

`<option value="mynum">One</option`

Now th:field maps the key (value attribute from option tag, in the example above "mynum") to the "treat" variable. But I want that the variable "treat" contains "One". How can I achieve this?

Thank you in advance!

**Comment by landsman** [comment]
what about this?

```html
<select class="form-select input-widget" id="treat" tabindex="31" th:field="*{treat}">
    <option value="">Please select...</option>
    <option th:each="v : ${@baba.getTreat()}" th:data-key="${v.key}" th:text="${v.value}" th:value="${v.value}"></option>
</select>
```

**Reply by pellerito** [reply]
Hi @landsman,
thank you for the response. I tried it, but unfortunately it doesnt work. In this case the treat variable stores neither the key nor the value. Honestly i doesnt understand the th:data-key attribute, it isnt a default thymeleaf attribute. Do you have another idea?
